aws/amazon-chime-sdk-js
# Simulcast optimizations

**Issue #:**
None

**Description of changes:**
Optimize for two party calling by leveraging servers functionality of direct metric feedback.  When there are only two video participants in a meeting, only the high simulcast stream will be enabled.  That one simulcast stream will then adjust appropriately to uplink and downlink network conditions.

Reduce simulcast switches by smoothing out downlink estimates and requiring larger drops in estimated bandwidth at lower bitrates before adjusting our target rate.

Modify requirements to meet probing successful and add timeout to ensure it does not get stuck in probing state.


**Testing**
- Verify correct number of active simulcast streams based on number of video participants and number of active video sources.
- Verify uplink policy reacts appropriately to changes in uplink bandwidth for each of the active simulcast layer configurations
- Verify downlink policy reacts as expected to downlink bandwidth changes in a variety of different network topologies.

1. Have you successfully run `npm run build:release` locally?
Yes
2. How did you test these changes?
Ran the above tests with the local browser demo as well as with the serverless demo.  
Added/ran additional unit tests to cover different scenarios
3. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
No
4. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
No

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
